### PR TITLE
Sprint gets reset to full on new game

### PR DESCRIPTION
**Pull request by Minetest-j45**
im not sure what the difference is between ```ctf_match.register_on_new_match(function())``` and ```ctf.register_on_new_game(function())``` so im not sure which i need to use(feel free to make suggestions if you know) but if this works, then great

### CODE REVIEW

File: mods/other/sprint/init.lua [hunk 1]
```diff
@@ -122,3 +122,17 @@ end)
 minetest.register_on_leaveplayer(function(player)
 	players[player:get_player_name()] = nil
 end)
+
+ctf_match.register_on_new_match(function()
+	local connected_players = minetest.get_connected_players()
+	for _, player in ipairs(connected_players) do
+		players[player:get_player_name()].stamina = STAMINA_MAX
```
Comment (Panquesito7) [inline mods/other/sprint/init.lua:129]:
```suggestion
		player[player:get_player_name()].stamina = STAMINA_MAX
```

File: mods/other/sprint/init.lua [hunk 2]
```diff
@@ -122,3 +122,17 @@ end)
...
+	end
+end)
+
+ctf.register_on_new_game(function()
+	local connected_players = minetest.get_connected_players()
+	for _, player in ipairs(connected_players) do
+		players[player:get_player_name()].stamina = STAMINA_MAX
```
Comment (Panquesito7) [inline mods/other/sprint/init.lua:136]:
```suggestion
		player[player:get_player_name()].stamina = STAMINA_MAX
```

**Comment by Minetest-j45** [comment]
wait what... no? im confused

**Comment by Minetest-j45** [comment]
see line 119, how they reset it on respawn... it is ```players[player:get_player_name()].stamina```, not ```player[player:get_player_name()].stamina```

File: mods/other/sprint/init.lua (hunk 2 above)
Comment (Panquesito7) [inline mods/other/sprint/init.lua:136]: Where is the variable/parameter `players` defined?

File: mods/other/sprint/init.lua (hunk 1 above)
Comment (Panquesito7) [inline mods/other/sprint/init.lua:129]: Same here.

File: mods/other/sprint/init.lua (hunk 2 above)
Comment (Minetest-j45) [reply mods/other/sprint/init.lua:136]: line 22

File: mods/other/sprint/init.lua (hunk 1 above)
Comment (Minetest-j45) [reply mods/other/sprint/init.lua:129]: same answer as above

**Comment by appgurueu** [comment]
Two things:
1. Is `register_on_new_game` needed? Why is `register_on_new_match` insufficient?
2. If both are needed, make your function - which is the same for both - a named local one and pass the same function to both instead of copying & pasting

**Comment by Minetest-j45** [comment]
See my first comment i made, when i opened this, as i said... I dont know the difference between them so i out both. If you know, feel free to change my pr

**Comment by Anthony-De** [comment]
@Minetest-j45 
`register_on_new_game` only ever used on server start.
You only need to use `register_on_new_match`.

File: mods/other/sprint/init.lua [hunk 3]
```diff
@@ -122,3 +122,17 @@ end)
...
+	end
+end)
```
Comment (Anthony-De) [inline mods/other/sprint/init.lua:138]:
```suggestion
minetest.after(0, function()
	ctf_match.register_on_new_match(function()
		for _, player in pairs(players) do
			player.stamina = STAMINA_MAX
		end
	end)
end)
```

**Review by Anthony-De — CHANGES_REQUESTED** [review]

**Comment by Anthony-De** [comment]
There is a little issue, the hudbar doesn't update till the player starts sprinting.

**Comment by Panquesito7** [comment]
> update till the player starts sprinting

That should be fixed in another PR, I think.

File: mods/other/sprint/init.lua (hunk 3 above)
Comment (appgurueu) [reply mods/other/sprint/init.lua:138]: `minetest.after(0, ...)` should not be used, use `minetest.register_on_mods_loaded(...)` instead. Why is it needed here?

Comment (Anthony-De) [reply mods/other/sprint/init.lua:138]: Because ctf_match isn't defined when this file runs.

**Comment by Minetest-j45** [comment]
is this (with AnthonyDE's changes) ready to merge?

**Comment by Minetest-j45** [comment]
oops. i accidentally closed this PR and didnt notice

**Review by Panquesito7 — APPROVED** [review]

**Pull request closed without merging**